### ISSUE TITLE
Difference in included version of SQLite between 3.2.1 and 3.1.1 

### ISSUE THREAD

**Issue by AndreSteenveld** [pawelsalawa/sqlitestudio#3466]
There seems to be a difference in the compiled version of sqlite for version 3.2.1 and 3.1.1 on OSX, I haven't looked at any other operating systems. In sqlitestudio version 3.1.1 the output of ```pragma compile_options;``` is:
```
COMPILER=clang-6.1.0
ENABLE_FTS3
ENABLE_FTS5
ENABLE_JSON1
ENABLE_RTREE
SYSTEM_MALLOC
THREADSAFE=1
```

The output of the same command in version 3.2.1 is:
```
COMPILER=clang-6.1.0
ENABLE_FTS3
ENABLE_RTREE
THREADSAFE=1
``` 

To confirm this I've installed the latest version using brew (```$ brew cask install sqlitestudio```) and downloaded 3.1.1 from the sqlitestudio web page (https://sqlitestudio.pl/files/sqlitestudio3/complete/macosx/).

The lack of the JSON1 extension breaks some of my views but I can imagine that it would break full text search for other people. It took me a while to figure out that this was the problem because my database seemed to work well in older versions, it was rather counter intuitive. 

As a work around I have installed version 3.1.1 on my system.

The low hanging fruit solution would seem to be including a version of sqlite which includes everything but the kitchen sink. But a better solution IMO would be to be able to specify your own binary for sqlite.

**Comment by tobiaswaltl** [comment]
@pawelsalawa Are there any updates on when this might get fixed?
After updating to Catalina SQLiteStudio 3.1.1 keeps crashing randomly. Hence, staying on 3.1.1 was no longer an option.

**Comment by pawelsalawa** [comment]
I'm not 100% sure this will work, but you can try:
Install 3.2.1 (if that works on Catalina), then install 3.1.1 in other place (or keep the version you have already installed), then copy libsqlite3*.dylib (or something similar - basically the SQLite3 binary) to SQLiteStudio 3.2.1, overriding its SQLite3 binary.

With a little bit of luck this may work.

**Comment by haiduong87** [comment]
Can I make my own version that has FTS5 enabled?
Is it hard?

**Comment by AndreSteenveld** [comment]
Working based on your suggestion I've created symbolic links a locally installed version of sqlite (installed using brew).

```bash
# SQLiteStudio is installed at /Applications/SQLiteStudio.app and I found the libsqlite3*.dylib file in the directory /Applications/SQLiteStudio.app/Contents/Frameworks.
# My local install of sqlite lives at: /usr/local/opt/sqlite/bin/sqlite3

# Goto the directory where the bins live
$ cd /Applications/SQLiteStudio.app/Contents/Frameworks

# Backup the existing libs
$ mv ./libsqlite3.dylib ./libsqlite3.dylib.old
$ mv ./libsqlite3.0.dylib ./libsqlite3.0.dylib.old

# Link to the newer libs
$ ln -Ts /usr/local/opt/sqlite/lib/libsqlite3.dylib ./libsqlite3.dylib
$ ln -Ts /usr/local/opt/sqlite/lib/libsqlite3.0.dylib ./libsqlite3.0.dylib
```

To validate that this works I connected to a in memory database and ran the query `select * from pragma_database_list();` which at first didn't work and after linking in the newest version of sqlite does. 🎉

**Comment by pawelsalawa** [comment]
@haiduong87 it's not that hard. You need to compile it and then replace the library with your own, just like @AndreSteenveld did above.

How to compile you can learn from here: https://superuser.com/questions/202640/how-to-install-sqlite-from-source-on-mac

**Comment by haiduong87** [comment]
After reading the issue carefully and make some tests.
I think my issues is about System.Data.Sqlite plugin.

Because I can connect to a sqlite3 database file, and perform fts5 queries without any problem.
But when I try to create a System.Dada.Sqlite and try the fts5 query, it said "no such module: fts5"

I need a way to perform these 2 commands with the plugin:
```
MyConnection.EnableExtensions(True)
MyConnection.LoadExtension("System.Data.SQLite.dll", "sqlite3_fts5_init")
```

https://stackoverflow.com/questions/37565423/sqlite-no-such-module-fts5-error-with-system-data-sqlite-dll-1-0-101-0

**Comment by pawelsalawa** [comment]
@haiduong87 unfortunately you cannot do it from SQLiteStudio evel, this has to be done from C++ level, so before compilation.

I created issue to track it (above).

**Comment by pawelsalawa** [comment]
Resolved with #3303. With CI/CD in place, package building is automated and unified - same compilation flags applied to SQLite on all 3 platforms:
```
                    -DSQLITE_ENABLE_UPDATE_DELETE_LIMIT \
                    -DSQLITE_ENABLE_DBSTAT_VTAB \
                    -DSQLITE_ENABLE_BYTECODE_VTAB \
                    -DSQLITE_ENABLE_COLUMN_METADATA \
                    -DSQLITE_ENABLE_EXPLAIN_COMMENTS \
                    -DSQLITE_ENABLE_FTS3 \
                    -DSQLITE_ENABLE_FTS4 \
                    -DSQLITE_ENABLE_FTS5 \
                    -DSQLITE_ENABLE_GEOPOLY \
                    -DSQLITE_ENABLE_JSON1 \
                    -DSQLITE_ENABLE_RTREE
```
